### PR TITLE
Destroy test_results after test_case is destroyed

### LINKED ISSUE #1197: Foreign key violation when uploading programming package
```The package could not be imported: PG::ForeignKeyViolation: ERROR: update or delete on table "course_assessment_question_programming_test_cases" violates foreign key constraint "fk_course_assessment_answer_bbb492885b1e3dca4433b8af8cb95906" on table "course_assessment_answer_programming_auto_grading_test_results" DETAIL: Key (id)=(12) is still referenced from table "course_assessment_answer_programming_auto_grading_test_results". : DELETE FROM "course_assessment_question_programming_test_cases" WHERE "course_assessment_question_programming_test_cases"."id" = $1```

This is either due to selecting a different language or trying to update the package after an answer has been attempted.

### PR DESCRIPTION
**Pull request by allenwq**
Follow #1206, this will fix #1197

### CODE REVIEW

File: app/models/course/assessment/question/programming_test_case.rb [hunk 1]
```diff
@@ -6,4 +6,9 @@ class Course::Assessment::Question::ProgrammingTestCase < ActiveRecord::Base
 
   belongs_to :question, class_name: Course::Assessment::Question::Programming.name,
                         inverse_of: :test_cases
+  has_many :test_results,
+           class_name: Course::Assessment::Answer::ProgrammingAutoGradingTestResult.name,
+           inverse_of: :nil,
```
Comment (fonglh) [inline app/models/course/assessment/question/programming_test_case.rb]: Does this need to be specified?

Comment (allenwq) [inline app/models/course/assessment/question/programming_test_case.rb]: Yes, otherwise will give exceptions

Comment (allenwq) [inline app/models/course/assessment/question/programming_test_case.rb]: Oh understand you wrongly, yes it need to be specified.

Comment (fonglh) [inline app/models/course/assessment/question/programming_test_case.rb]: As in, need to specify that it's `inverse_of: :test_cases` ? or something along those lines.

**Pull request merged**